src/esp/assets/ResourceManager.cpp

```diff
@@ -1332,8 +1343,11 @@ void ResourceManager::addComponent(const MeshMetaData& metaData,
 
   // Recursively add children
   for (auto& child : meshTransformNode.children) {
-    addComponent(metaData, node, lightSetup, drawables, child);
+    std::vector<scene::SceneNode*> childNodes =
+        addComponent(metaData, node, lightSetup, drawables, child);
```

Comment:
(Probably an unnecessary optimization given that there's a lot more allocations elsewhere anyway, but since I already went through the train of though I'll leave it here -- it's not a big change after all.)

- - -

I could imagine an `addComponentInternal(...., std::vector<scene::SceneNode*>& newNodes)` that avoids the extra copies/allocations on recursion:

```suggestion
    addComponentInternal(metaData, node, lightSetup, drawables, newNodes);
```

and the public `addComponent()` would then look simply like this:

```cpp
std::vector<scene::SceneNode*> ResourceManager::addComponent(
    const MeshMetaData& metaData,
    scene::SceneNode& parent,
    const Mn::ResourceKey& lightSetup,
    DrawableGroup* drawables,
    const MeshTransformNode& meshTransformNode) {
  std::vector<scene::SceneNode*> newNodes;
  addComponentInternal(metaData, parent, lightSetup, drawables, meshTransformNode, newNodes);
  return newNodes;
}
```